src/awssdk_driver.rs

```diff
@@ -21,10 +19,35 @@ use aws_sdk_qldbsession::{
 use aws_smithy_client::bounds::SmithyConnector;
 use aws_smithy_client::erase::DynConnector;
 use aws_smithy_client::Client;
-use aws_types::app_name::AppName;
+use aws_smithy_http::{middleware::MapRequest, operation};
+use aws_smithy_http_tower::map_request::MapRequestLayer;
+use http::header::HeaderName;
+use http::Uri;
+use tower::ServiceBuilder;
 
 use crate::{error, settings::Environment};
 
+#[derive(Clone, Debug)]
+struct UserAgent;
+
+impl MapRequest for UserAgent {
```

Comment:
Can you please write a test for this that shows that, at least, the code doesn't panic? (Given it uses unwrap.)

Reply:
If you accept the `from_static` changes, you can ignore this comment.

Reply:
thanks, changes accepted